src/cursor.c

```diff
@@ -464,8 +461,9 @@ PyObject *cursor_fetchall(CursorObject *cursor, PyObject *args) {
         cursor->status = CURSOR_STATUS_READY;
         break;
       } else if (fetch_status == 1) {
-        if (PyList_Append(results, row) < 0) {
-          Py_DECREF(row);
+        int append_result = PyList_Append(results, row);
+        Py_DECREF(row);
```

Comment:
It seems the common problem was that everyone assumed that successful calls to the functions like `PyList_Append` would give up the ownership of that object and automatically handle the ref counting stuff.
A good thing to have in mind for future development!